src/app/admin/contacts-list/contacts-list.component.html

```diff
@@ -19,15 +20,23 @@
       <div class="col-sm-10 col-sm-offset-1">
         <div class="panel panel-default">
           <div class="panel-body">
-            <div class="pull-right">
-              <div class="btn-group">
-                <button type="button" class="btn btn-success btn-filter" (click)="showNew()">New</button>
-                <button type="button" class="btn btn-default btn-filter" (click)="showAll()">All</button>
-              </div>
-            </div>
+              <form class="navbar-form search-form" role="search">
+                <div class="btn-group">
+                <button type="button" class="btn btn-default btn-filter" (click)="showNew()">New</button>
+                <button type="button" class="btn btn-primary btn-filter black-styled" (click)="showAll()">All</button>
+              </div> 
+                <div class="focus-container" tabindex="-1">
+                  <div class="input-group add-on">
+                    <input type="text" class="form-control" placeholder="Search" name="srch-term" id="srch-term" [(ngModel)]="queryString">
+                    <div class="input-group-btn">
+                      <button class="btn btn-default" type="submit"><i class="glyphicon glyphicon-search"></i></button>
+                    </div>
+                  </div>
+                </div>
+              </form>
             <table class="table table-filter">
               <tbody>
-                <tr *ngFor="let key of messages | messageStatus:fullList; let i = index">
+                <tr *ngFor="let key of messages | messageStatus:fullList | searchNews:queryString:'body'; let i = index" [class.newMessage]="key.new">
```

Comment:
Why `key` instead of `message`? And don't you want to make this filtering on component layer? It's not pipe's job